MIGRATING_v2_to_v3.md

```diff
@@ -0,0 +1,236 @@
+# Okta Java Management SDK Migration Guide
+ 
+This SDK uses semantic versioning and follows Okta's 
+[library version policy](https://developer.okta.com/code/library-versions/). 
+In short, we do not make breaking changes unless the major version changes!
+
+## Migrating from 2.x.x to 3.0.0
+
+Version 3.0.0 of this SDK introduces a number of breaking changes from previous versions. 
+In addition to many new classes/interfaces, some existing classes/interfaces are no longer backward compatible due to method renaming and signature changes.
+
+### Package `com.okta.sdk.client.Client`
+
+Below method has been added.
+- `void deleteIdentityProviderKey(String keyId)`
```

Comment:
In general, a migration guide doesn't need to add new features, it needs to list what features have changed, and how it will affect existing users

Reply:
I've added a global explanation of what was changed [here](https://github.com/okta/okta-sdk-java/pull/495/commits/76e127214519f4743426775a5a58f669a03805e0)